src/elfpy/pricing_models/hyperdrive.py

```diff
@@ -112,8 +112,17 @@ def calc_in_given_out(
         d_bonds = out_amount * (1 - normalized_time)
         d_shares = d_bonds / share_price
 
-        # TODO: Verify that this is needed.
-        market_state = copy.copy(market_state)
+        # this is needed to not modify the actual market state, as this is a mock only
+        market_state = copy.copy(
```

Comment:
I don't think you need `copy.copy` if you're instantiating a new object.

Reply:
true, I was doing it without, then half-reverted. i'll revert fully.